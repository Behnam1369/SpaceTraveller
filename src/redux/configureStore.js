import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import missionReducer from './missions';
import { rocketsReducer } from './rockets';

const rootReducer = combineReducers({
  rockets: rocketsReducer,
  missions: missionReducer,
});

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
