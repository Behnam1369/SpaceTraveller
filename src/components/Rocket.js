import { React } from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { addReserv, removeReserv } from '../redux/rockets';

function Rocket(props) {
  const {
    name, description, image, el,
  } = props;
  const dispatch = useDispatch();
  const reservButton = (element) => {
    if (element.reserved === false) {
      return <button className="reserv-btn add-btn" type="button" data-testid="quit" onClick={() => { dispatch(addReserv(element.id)); }}>Reserve now!</button>;
    }
    return <button className="reserv-btn rm-btn" type="button" onClick={() => { dispatch(removeReserv(element.id)); }}>Remove reservation</button>;
  };

  return (
    <div className="rocket-card">
      <div className="rocket-img">
        <img src={image} className="rocket-img-src" alt="rocket-img" />
      </div>
      <div className="rocket-content">
        <h2>{name}</h2>
        <p>
          {el.reserved === true ? <span className="reserve-span" data-testid="teal">reserved</span> : null}
          {description}
        </p>
        {reservButton(el)}
      </div>
    </div>
  );
}

Rocket.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  el: PropTypes.objectOf(PropTypes.shape()).isRequired,
};

export default Rocket;
