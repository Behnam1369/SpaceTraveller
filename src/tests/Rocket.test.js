import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rocket from '../components/Rocket';
import '@testing-library/jest-dom';
import configureStore from '../redux/configureStore';

describe('Testing Mission', () => {
  it('Testing Mission component snapshot', () => {
    const element = {
      id: 'key1',
      name: 'TEST ROCKET',
      description: 'TEST DESCRIPTION',
      image: 'https://imgur.com/DaCfMsj.jpg',
      reserved: true,
    };
    const testRenderer = TestRenderer.create(
      <Provider store={configureStore}>
        <Rocket
          key="key1"
          name="key1"
          description="A Sample rocket"
          image="https://imgur.com/DaCfMsj.jpg"
          el={element}
        />
      </Provider>,
    ).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
});
