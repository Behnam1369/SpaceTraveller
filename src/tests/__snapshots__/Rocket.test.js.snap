// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Mission Testing Mission component snapshot 1`] = `
<div
  className="rocket-card"
>
  <div
    className="rocket-img"
  >
    <img
      alt="rocket-img"
      className="rocket-img-src"
      src="https://imgur.com/DaCfMsj.jpg"
    />
  </div>
  <div
    className="rocket-content"
  >
    <h2>
      key1
    </h2>
    <p>
      <span
        className="reserve-span"
        data-testid="teal"
      >
        reserved
      </span>
      A Sample rocket
    </p>
    <button
      className="reserv-btn rm-btn"
      onClick={[Function]}
      type="button"
    >
      Remove reservation
    </button>
  </div>
</div>
`;
